go/vt/vtgate/engine/route.go

```diff
@@ -588,11 +599,11 @@ func (route *Route) paramsAnyShard(vcursor VCursor, bindVars map[string]*querypb
 }
 
 func (route *Route) paramsSelectEqual(vcursor VCursor, bindVars map[string]*querypb.BindVariable) ([]*srvtopo.ResolvedShard, []map[string]*querypb.BindVariable, error) {
-	key, err := route.Values[0].ResolveValue(bindVars)
+	value, err := route.Value.ResolveValue(bindVars)
 	if err != nil {
 		return nil, nil, err
 	}
-	rss, _, err := resolveShards(vcursor, route.Vindex, route.Keyspace, []sqltypes.Value{key})
+	rss, _, err := resolveShards(vcursor, route.Vindex, route.Keyspace, []sqltypes.Value{value})
```

Comment:
can we do a `values, err := route.Value.ResolveList(bindVars)`, and pass `values` to `resolveShards` instead of casting `value` to a slice of `sqltypes.Value`?

Reply:
ResolveList is only for tuple values, equal value will not be a tuple.